ui/src/logs/components/expandable_message/ExpandableMessage.tsx

```diff
@@ -16,48 +23,74 @@ interface Props {
   searchPattern: string
 }
 
+@ErrorHandling
 export class ExpandableMessage extends Component<Props, State> {
+  private containerRef: React.RefObject<HTMLDivElement>
+
   constructor(props) {
     super(props)
+    this.containerRef = React.createRef()
     this.state = {
       expanded: false,
     }
   }
 
   public render() {
-    const {notify, searchPattern} = this.props
-    const formattedValue = `${this.props.formattedValue}`
-    const trimmedValue = formattedValue.trimLeft()
+    return (
+      <div
+        onClick={this.handleClick}
+        className="expandable--message"
+        ref={this.containerRef}
+      >
+        {this.message}
+        {this.expandedMessage}
+      </div>
+    )
+  }
+
+  private get message(): JSX.Element {
+    const {notify, searchPattern, formattedValue} = this.props
+    const valueString = `${formattedValue}`
+    const trimmedValue = valueString.trimLeft()
 
     return (
-      <ClickOutside onClickOutside={this.handleClickOutside}>
-        <div onClick={this.handleClick} className="expandable--message">
-          <div className="expandable--text">
-            <LogsMessage
-              formattedValue={trimmedValue}
-              notify={notify}
-              searchPattern={searchPattern}
-            />
-          </div>
-          <div className={this.isExpanded}>
-            {this.closeExpansionButton}
-            <LogsMessage
-              formattedValue={formattedValue}
-              notify={notify}
-              searchPattern={searchPattern}
-            />
-          </div>
-        </div>
-      </ClickOutside>
+      <LogsMessage
+        formattedValue={trimmedValue}
+        notify={notify}
+        searchPattern={searchPattern}
+      />
     )
   }
 
-  private get isExpanded() {
+  private get expandedMessage() {
     const {expanded} = this.state
-    if (expanded) {
-      return 'expanded--message'
-    } else {
-      return 'collapsed--message'
+    const container = document.getElementById('expanded-message-container')
+
+    if (!expanded || !this.containerRef.current) {
+      return null
+    }
+
+    if (this.containerRef.current) {
```

Comment:
Do you still need this with the early return null?

Reply:
ahhh good catch